File: oo-inheritance.swift
class Vehicle {
  func move() {
    print("vehicle move")
  }
}

class Car : Vehicle {

  func foo() {
    print("car foo")
  }

}

class Bike : Vehicle {

  override func move() {
    print("special bike move")
  }

  func foo() {
    print("bike foo")
  }
}

let car = Car()
let bike = Bike()

car.move()
bike.move()

car.foo()
bike.foo()

File: oo-inheritance.rb
class Vehicle

  def move
    puts "vehicle move"
  end

end

class Car < Vehicle

  def foo
    puts "car foo"
  end

end

class Bike < Vehicle

  # overwrite
  def move
    puts "special bike move"
  end

  def foo
    puts "bike foo"
  end

end

car = Car.new
bike = Bike.new

car.move()
bike.move()

car.foo()
bike.foo()
